doc/Coding_guidelines.md

```diff
@@ -191,9 +191,11 @@ Feel free to improve this document. Bring more examples and rules. But please tr
 
 10. Use unsigned types whenever it possible. Unsigned overflow is well defined behavior while signed is not.
 
+11. Avoid usage of local ```static``` variables because it might bring unexpectable overhead. Since C++11 initialization of local ```static``` variables must be thread safe, so compiler generates additional code to provide it.
```

Comment:
For me, this is too strong statement, because in some cases local `static` is better than namespace scope's or even class member ones just because of more local scope. I'd rather re-phrase it something like
"Note that usage of local ```static``` variables because it might bring unexpectable overhead"

Reply:
@dmitry-gurulev I agree that it is not 100% rule and violation with a good reason is allowed. But as for me usage of local static variables shouldn't be a wide practice. Moreover, your concern is covered by statement
```
These tips shouldn't be considered as strict rules, they are just good piece of advice on how to implement your solution in modern C++. But please be sure to know recommended way before implementing something different. Violations of these rules must be committed consciously, otherwise better solution is most likely just to follow guidelines.
```

Reply:
OK, agreed w/ that interpretatin